docker/stan-wasm-server/logic/locking.py

```diff
@@ -0,0 +1,40 @@
+import os
+from random import seed, choices
+from string import ascii_letters
+
+def get_nonce():
+    seed(os.getpid())
+    return str.join('', choices(ascii_letters, k=20))
+
+
+def _get_compilation_lockfile_name(model_dir: str):
+    return os.path.join(model_dir, "running.txt")
+
+
+def acquire_compilation_lock(model_dir: str, nonce: str):
+    lock_name = _get_compilation_lockfile_name(model_dir)
+    try:
+        with open(lock_name, 'x') as file:
+            file.write(nonce)
+        with open(lock_name, 'r') as file:
+            contents = file.read()
+            assert contents == nonce
```

Comment:
`assert` turns into a no-op under certain runtime configurations and should probably be avoided.

Reply:
Yeah, I'm... not sure why I wasn't just doing a comparison there from the beginning.